internal/kafka/internal/services/kafka.go

```diff
@@ -1376,3 +1421,21 @@ func (k *kafkaService) IsQuotaEntitlementActive(kafkaRequest *dbapi.KafkaRequest
 
 	return quotaService.IsQuotaEntitlementActive(kafkaRequest)
 }
+
+type ClusterSizeCount struct {
+	SizeId string
+	Count  int32
+}
+
+func (k *kafkaService) GetKafkaSizeCountOfEnterpriseCluster(clusterID string) ([]*ClusterSizeCount, error) {
+	dbConn := k.connectionFactory.New()
+	var sizeCountPerCluster []*ClusterSizeCount
+	if err := dbConn.Model(&dbapi.KafkaRequest{}).
+		Select("size_id, count(1) as Count").
```

Comment:
might be good to also return the kafka instance type as the streaming unit notion is dependent on this